### PR TITLE
Ft-google-facebook-auth-182201570

### PR DESCRIPTION
**Pull request by Francoise02**
**What does this PR do?**
Allow users to log in with their Google and Facebook

**How should this be manually tested?**

1. Create Google Client and Secret IDs: you can follow this [tutorial](https://www.youtube.com/watch?v=xH6hAW3EqLk) for guidance.

2. Create Facebook Client and secret IDs. You can follow this [tutorial](https://www.youtube.com/watch?v=jMVlSDU-6l4) to do so.

3. Follow the.env.example file to create your .env file and add your credentials

4. Run migration
5. Start the server

Check these endpoints:

`GET : auth/google`
`GET : auth/facebook`

**Any background context you want to provide?**

You need to have both the endpoint path and a correct callback_url to successfully test this PR.

**What are the relevant pivotal tracker stories?**
https://www.pivotaltracker.com/story/show/182201570

### CODE REVIEW

**Review by Julienish14 — APPROVED** [review]

**Review by Julienish14 — DISMISSED** [review]

**Review by Julienish14 — DISMISSED** [review]

**Review by uwishema11 — DISMISSED** [review]

File: .env.example [hunk 1]
```diff
@@ -20,6 +20,18 @@ JWT_KEY=newUserKey
 JWT_SECRET=secret
 JWT_EXPIRES_IN=time
 JWT_COOKIE_EXPIRES_IN=time 
+REFRESH_TOKEN_KEY=Jwt_Refresh_key
+
+# Google api connection
+GOOGLE_CLIENT_ID= 1029104825308-ukbvcpdqa58u468gj1dcf1erjpag2ipd.apps.googleusercontent.com
+GOOGLE_CLIENT_SECRET= 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
```
Comment (didasmbalanya) [inline .env.example]: remove these values from github. they are private.

File: .env.example [hunk 2]
```diff
@@ -20,6 +20,18 @@ JWT_KEY=newUserKey
...
+GOOGLE_CALLBACK_URL = http://localhost:3000/auth/google/callback
+
+
+# Facebook api connection
+FACEBOOK_CLIENT_ID= 1393150511168772
+FACEBOOK_CLIENT_SECRET= 0b7fb4122db92dadc84325e4242c87f4
```
Comment (didasmbalanya) [inline .env.example]: private values. Should never be pushed to Github

File: src/controllers/googleFacebook.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+ */
+
+export const userToken = async (req, res) => {
+  const user = {
+    id: req.user.id,
+    firstname: req.user.firstName,
+    lastname: req.user.lastName,
+    email: req.user.email
+  };
+  console.log(user)
```
Comment (didasmbalanya) [inline src/controllers/googleFacebook.js]: remove console

File: src/controllers/googleFacebook.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  const token = await storeToken(user);
+  console.log(token)
```
Comment (didasmbalanya) [inline src/controllers/googleFacebook.js]: remove

File: src/routes/googleFacebook.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+/* eslint-disable */
```
Comment (didasmbalanya) [inline src/routes/googleFacebook.js]: Why are you disabling eslint? Please remove this comment

File: src/controllers/index.js [hunk 6]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable */
```
Comment (didasmbalanya) [inline src/controllers/index.js]: remove this

File: src/controllers/googleFacebook.js [hunk 7]
```diff
@@ -0,0 +1,119 @@
+/* eslint-disable */
```
Comment (didasmbalanya) [inline src/controllers/googleFacebook.js]: remove

File: src/database/migrations/20220321064904-authorizedUser.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
+/* eslint-disable */
```
Comment (didasmbalanya) [inline src/database/migrations/20220321064904-authorizedUser.js]: remove

File: src/database/migrations/20220321064904-authorizedUser.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE
+      }
+    });
+  },
+  async down(queryInterface, Sequelize) {
+    await queryInterface.dropTable('authorizedUsers');
+  }
+};
```
Comment (didasmbalanya) [inline src/database/migrations/20220321064904-authorizedUser.js:29]: You do not need this table

Comment (didasmbalanya) [reply src/database/migrations/20220321064904-authorizedUser.js:29]: We use JWTs to identify authorized users

File: src/database/migrations/20220605123824-create-user.js [hunk 10]
```diff
@@ -4,25 +4,43 @@ module.exports = {
   async up(queryInterface, Sequelize) {
     await queryInterface.createTable('Users', {
       id: {
+        type: Sequelize.UUID,
+        defaultValue: Sequelize.UUIDV4,
         allowNull: false,
-        autoIncrement: true,
-        primaryKey: true,
-        type: Sequelize.INTEGER
+        primaryKey: true
```
Comment (didasmbalanya) [inline src/database/migrations/20220605123824-create-user.js:9]: do not touch old migration files.
Write a new migration file and do whatever updates!

File: src/database/migrations/20220605123824-create-user.js [hunk 11]
```diff
@@ -4,25 +4,43 @@ module.exports = {
...
+        type: Sequelize.STRING,
+        allowNull: true
+      },
+     
+      isVerified: {
+        type: Sequelize.BOOLEAN,
+        defaultValue: false
       },
       role: {
         type: Sequelize.ENUM,
```
Comment (didasmbalanya) [inline src/database/migrations/20220605123824-create-user.js]: do all this changes in a separate migration file

File: src/database/models/authorized.js [hunk 12]
```diff
@@ -0,0 +1,29 @@
+/* eslint-disable*/
```
Comment (didasmbalanya) [inline src/database/models/authorized.js]: remove

File: src/database/models/authorized.js [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+    },
+    {
+      sequelize,
+      modelName: 'authorizedUser',
+      tableName: 'authorizedUsers'
+    }
+  );
+  return authorizedUser;
+};
+
```
Comment (didasmbalanya) [inline src/database/models/authorized.js:27]: what is this 'authorizedUser' model for ?

File: src/database/models/index.js [hunk 14]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable*/
```
Comment (didasmbalanya) [inline src/database/models/index.js]: remove

File: src/database/seeders/user.js [hunk 15]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable*/
```
Comment (didasmbalanya) [inline src/database/seeders/user.js]: remove

File: src/index.js [hunk 16]
```diff
@@ -1,4 +1,4 @@
-
+/* eslint-disable */
```
Comment (didasmbalanya) [inline src/index.js]: remove

File: src/routes/index.js [hunk 17]
```diff
@@ -1,8 +1,10 @@
+/* eslint-disable */
```
Comment (didasmbalanya) [inline src/routes/index.js]: remove

File: src/services/googleFacebook.js [hunk 18]
```diff
@@ -0,0 +1,43 @@
+/* eslint-disable */
```
Comment (didasmbalanya) [inline src/services/googleFacebook.js]: remove

File: src/services/googleFacebook.js [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+    const userData = {
+      id: user.id,
+      email: user.email
+    };
+    const accessToken = await generateAccessToken(userData);
+    const refreshToken = await generateRefreshToken(userData);
+    await updateOrCreate(
+      authorizedUser,
+      // { user_id: userData.id },
+      // { user_id: userData.id, refreshToken }
```
Comment (didasmbalanya) [inline src/services/googleFacebook.js]: remove commented code if it adds no value

File: src/services/userServices.js [hunk 20]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable */
```
Comment (didasmbalanya) [inline src/services/userServices.js]: remove

File: src/utils/helpers/generateToken.js [hunk 21]
```diff
@@ -1,25 +1,31 @@
+/* eslint-disable */
```
Comment (didasmbalanya) [inline src/utils/helpers/generateToken.js]: remove

**Review by didasmbalanya — CHANGES_REQUESTED** [review]

**Review by kalisaNkevin — DISMISSED** [review]

**Review by ishimweeli — DISMISSED** [review]
running well on my end

**Review by Julienish14 — DISMISSED** [review]

**Review by Julienish14 — APPROVED** [review]

**Review by munyanezaarmel — APPROVED** [review]

**Review by Julienish14 — APPROVED** [review]

**Review by kalisaNkevin — APPROVED** [review]

**Review by uwishema11 — APPROVED** [review]

**Review by Ntare22 — APPROVED** [review]
LGTM

**Pull request merged**